packages/iov-core/src/transactionencoder.spec.ts

```diff
@@ -0,0 +1,159 @@
+import {
+  Address,
+  Algorithm,
+  ChainId,
+  isPublicIdentity,
+  PublicKeyBytes,
+  SendTransaction,
+  TokenTicker,
+} from "@iov/bcp";
+import { Encoding } from "@iov/encoding";
+
+import { TransactionEncoder } from "./transactionencoder";
+
+const { fromJson, toJson } = TransactionEncoder;
+
+describe("TransactionEncoder", () => {
+  describe("toJson", () => {
+    it("works for numbers", () => {
+      expect(toJson(0)).toEqual(0);
+      expect(toJson(1)).toEqual(1);
+      expect(toJson(-1)).toEqual(-1);
+      expect(toJson(Number.POSITIVE_INFINITY)).toEqual(Number.POSITIVE_INFINITY);
+      expect(toJson(Number.NaN)).toEqual(Number.NaN);
+    });
+
+    it("works for booleans", () => {
+      expect(toJson(true)).toEqual(true);
+      expect(toJson(false)).toEqual(false);
+    });
+
+    it("works for null", () => {
+      expect(toJson(null)).toEqual(null);
+    });
+
+    it("works for arrays", () => {
+      expect(toJson([])).toEqual([]);
+      expect(toJson([1, 2, 3])).toEqual([1, 2, 3]);
+      expect(toJson([1, [2, 3]])).toEqual([1, [2, 3]]);
+    });
+
+    it("works for dicts", () => {
+      expect(toJson({})).toEqual({});
+      expect(toJson({ foo: 1 })).toEqual({ foo: 1 });
+      expect(toJson({ foo: 1, bar: 2 })).toEqual({ foo: 1, bar: 2 });
+      expect(toJson({ foo: { bar: 1 } })).toEqual({ foo: { bar: 1 } });
+    });
+
+    it("fails for unsupported objects", () => {
+      expect(() => toJson(() => 0)).toThrowError(/Cannot encode type to JSON/i);
```

Comment:
How about storing the regex in a variable so it's clear they all fail with the same message?